### ISSUE TITLE
Cmpatible loss function

### ISSUE THREAD

**Issue by bva1986** [auspicious3000/autovc#16]
Hello

To reproduce results of AutoVC we used following loss function implementation on PyTorch
```
num_steps = 100000
G = Generator(32,256,512,32).to(device)
...
criterion_recon = nn.MSELoss().to(device)
criterion_recon0 = nn.MSELoss().to(device)
criterion_content = nn.L1Loss().to(device)
optimizer = optim.Adam(G.parameters(), lr=1e-4)
G.train()
...
for i in range(1, num_steps + 1):
    ...
    optimizer.zero_grad()
    X1rt, X1r, C1 = G(X1, S1, S1)
    #with torch.no_grad(): # Should we calculate C1r with grad?
    #    C1r = G(X1r[:, 0, :, :], S1, None)
    C1r = G(X1r[:, 0, :, :], S1, None)
    L_recon = criterion_recon(X1r[:, 0, :, :], X1)
    L_recon0 = criterion_recon0(X1rt[:, 0, :, :], X1)
    L_content = criterion_content(C1r, C1)
    loss = L_recon + 1.0*L_recon0 + 1.0*L_content
    loss.backward()
    optimizer.step()
```
But we did not achieved comparable voice quality and loss value around 1e-3   according issue [whats your final loss and final learning rate? ](https://github.com/auspicious3000/autovc/issues/10)
One of supposed reason is that we used inapropriate loss function implementation.
Is our loss function implementation compatible with that you used?

BR

**Comment by auspicious3000** [comment]
yes, your loss function looks right

**Comment by bva1986** [comment]
Should we backpropogate Encoder->Decoder->Encoder or just Encoder when minimize loss C1r-C1 ?
```
C1r = G(X1r[:, 0, :, :], S1, None)
L_content = criterion_content(C1r, C1)
```
or
```
C1r = G(X1r[:, 0, :, :].detach(), S1, None)
L_content = criterion_content(C1r, C1)
```

**Comment by auspicious3000** [comment]
The 1st one

**Comment by bva1986** [comment]
According to AutoVC presentation on ICML-2019 loss function is:
L =  Lrecon + lambda * Lcontent.
But according to paper https://arxiv.org/abs/1905.05879 loss function is 
L = Lrecon + mu * Lrecon0 + lambda*Lcontent.
Which of them is more preferable?

**Comment by auspicious3000** [comment]
The 2nd one

**Comment by bva1986** [comment]
Ok, thanks
